src/number.rs

```diff
@@ -63,6 +63,27 @@ impl Number {
         }
     }
 
+    /// Construct a new `Number` from parts, stripping unnecessary trailing zeroes.
+    /// This can't create a NaN value.
+    ///
+    /// ```
+    /// # use json::number::Number;
+    /// let one = Number::from_parts_strip(true, 1000, -3);
+    /// let (positive, mantissa, exponent) = one.as_parts();
+    ///
+    /// assert_eq!(true, positive);
+    /// assert_eq!(1, mantissa);
+    /// assert_eq!(0, exponent);
+    /// ```
+    #[inline]
+    pub fn from_parts_strip(positive: bool, mut mantissa: u64, mut exponent: i16) -> Self {
```

Comment:
I reckon this could be just `from_parts`, while the original `from_parts` could be renamed to `from_parts_unchecked` (to stay in line with stdlib naming). Changing the unchecked version to `pub unsafe fn` as a deterrent against using it outside of the parser / interger converts would be a bonus.